gulpfile.js

```diff
@@ -208,6 +209,47 @@ gulp.task('manifest:testing', function () {
 
   // Exclude chromereload script in production:
     .pipe(jsoneditor(function (json) {
+      json.permissions = [...json.permissions, 'webRequestBlocking', 'http://localhost/8889']
+      return json
+    }))
+
+    .pipe(gulp.dest('./dist/', { overwrite: true }))
+})
+
+const scriptsToExcludeFromBackgroundDevBuild = {
+  'bg-libs.js': true,
+}
+
+gulp.task('manifest:testing-local', function () {
```

Comment:
Interesting - so this task is only used for `test:copy`, which is only used by `dev:test`, which itself is only called from the `start:test` npm script.

What is the `start:test` script used for anyway? Why does it need the `localhost/8889` permission? It looks like it's basically the same as the dev build except that `IN_TEST` is set to true. I'm not sure why it'd need a special permission.

Reply:
`start:test` shares `json.permissions = [...json.permissions, 'webRequestBlocking', 'http://localhost/8889']` with the `manifest:testing` task, which is run with `build:test`. Whenever we are running tests, we need these permissions. `'webRequestBlocking'` is needed for tests so that we can prevent certain network requests. `'http://localhost/8889'` is being added with this PR so that we can make certain http requests to a local server.

Like `start`, `start:test` intentionally excludes `bg-libs.js` from the background files listed in the manifest... because `start` doesn't build that file (I can't remember why that decision was made).

`start:test` itself is useful when you want a watched build of metamask running while working on e2e tests, or if you want to run metamask in the browser with `IN_TEST` set to true

Reply:
Right - I had thought at the time that the `localhost` permission was only added in this task - I hadn't realized it was in `manifest:testing` as well. I had thought that was the reason for creating `manifest:testing-local` in the first place.

But rather it looks like the reason was so that `bg-libs` could be excluded.

This seems fine, though I remain of the opinion that we'd be better off splitting those background libraries out in development and testing just as we do in production. Particularly as the gulpfile grows more complex to accommodate this.